thunder/tests/distributed/test_ddp.py

```diff
@@ -1501,21 +1507,34 @@ def _test_ddp_transformer_engine_llama_sanity(input_data):
         fwd_exec_trace = thunder.last_traces(jit_model)[-1]
         bwd_exec_trace = thunder.last_backward_traces(jit_model)[-1]
 
-        # Verify that the first te_linear in fwd_exec_trace is the
-        # last one in bwd_exec_tarce.
-        # We verify that by managing the `ctx` (CollectionProxy) output by `te_linear` which is
-        # passed to backward.
-        # As CollectionProxy don't implement __eq__, we verify them by name.
-        first_ctx_name = None
-        for bsym in fwd_exec_trace.bound_symbols:
-            if bsym.sym.name.startswith("te_linear"):
-                first_ctx_name = bsym.output[1].name
-                break
-
-        for bsym in reversed(bwd_exec_trace.bound_symbols):
-            if bsym.sym.name.startswith("te_functional"):
-                assert first_ctx_name == bsym.args[-1].name, (first_ctx_name, bsym.args[-1].name)
-                break
+        if TE_VERSION_1_6_PLUS:
+            # Verify that the symbol to sync backward
+            # fp8 metadata is present in backward trace.
+            found_bwd_sync_symbol = False
+            for bsym in reversed(bwd_exec_trace.bound_symbols):
+                if bsym.sym.id == te_sync_fp8_meta_bwd.id:
+                    found_bwd_sync_symbol = True
+                    break
+
+            if not found_bwd_sync_symbol:
+                raise RuntimeError("Backward sync symbol not found.")
```

Comment:
nit: neat python magic https://book.pythontips.com/en/latest/for_-_else.html

```suggestion
            for bsym in reversed(bwd_exec_trace.bound_symbols):
                if bsym.sym.id == te_sync_fp8_meta_bwd.id:
                    break
            else:
                raise RuntimeError("Backward sync symbol not found.")
```

Reply:
Nice, thank you!